File: 2.Pattern Part 2/DiamondPattern.java
import java.util.*;

public class DiamondPattern {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        int n = sc.nextInt();

        //outer loop ->  upper triangle
        for(int i=1; i<=n; i++){
            //inner loop -> spaces
            for(int j = 1; j<=n-i;j++){
                System.out.print(" ");
            }

            //inner loop -> stars
            for(int j=1; j<=(2*i)-1; j++){
                System.out.print("*");
            }

            System.out.println();
        }
        
        //outer loop -> lower triangle
        for(int i=n; i>=1; i--){
            //inner loop -> spaces
            for(int j=1; j<=n-i; j++){
                System.out.print(" ");
            }

            //inner loop -> stars
            for(int j=1; j<=2*i-1; j++){
                System.out.print("*");
            }
            System.out.println();
        }
        sc.close();
    }
}

File: PythonPatterns/PatternPart2/DiamondPattern.py
# Diamond Pattern: Prints a symmetric diamond-shaped star pattern

def diamond_pattern(n):
    # Upper triangle of the diamond
    for i in range(1, n + 1):
        # Print leading spaces
        for j in range(1, n - i + 1):
            print(" ", end="")
        # Print stars
        for j in range(1, 2 * i):
            print("*", end="")
        print()  # Newline after each row
    
    # Lower triangle of the diamond
    for i in range(n, 0, -1):
        # Print leading spaces
        for j in range(1, n - i + 1):
            print(" ", end="")
        # Print stars
        for j in range(1, 2 * i):
            print("*", end="")
        print()  # Newline after each row


def main():
    """
    Main function to take user input and handle invalid cases.
    """
    try:
        n = int(input("Enter number of rows for diamond pattern: "))
        diamond_pattern(n)
    except ValueError:
        print("Please enter a valid integer.")


# Program execution starts here
main()


'''
Output-: 

    *
   ***
  *****
 *******
*********
*********
 *******
  *****
   ***
    *
    
# Cross check with Invalid Input

Enter number of rows for diamond pattern: g
Please enter a valid integer.

'''